kernel/src/util/timer.rs

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MPL-2.0
+use alloc::{boxed::Box, string::ToString, sync::Arc};
+use core::time::Duration;
+
+use ostd::{
+    orpc::{
+        framework::{
+            errors::RPCError,
+            notifier::{Notifier, NotifierOQueues},
+        },
+        oqueue::OQueueRef,
+        orpc_impl, orpc_server,
+    },
+    sync::WaitQueue,
+};
+use snafu::Whatever;
+
+use crate::prelude::WaitTimeout;
+
+/// TimerServer periodically produces a notification with a configurable frequency.
+#[orpc_server(Notifier)]
+pub struct TimerServer {
+    freq: Duration,
+}
+
+#[orpc_impl]
+impl Notifier for TimerServer {
+    fn notify(&self) -> Result<(), RPCError> {
+        if self.notification_oqueue().produce(()).is_err() {
+            Err(RPCError::Panic {
+                message: "Could not produce into notification_oqueue".to_string(),
+            })
```

Comment:
`RPCError::Panic` should only be generated when the server panics.  If you actually want to crash the server, you can actually panic here. Or you can use the `kernel` Error. It has some catch all values.

(I'm going to rework the error types soon, but let's not get too sloppy.)